src/BuildScriptGenerator/Node/NodeVersionChecker.cs

```diff
@@ -6,26 +6,33 @@
 using System.Collections.Generic;
 using System.Linq;
 using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Oryx.BuildScriptGenerator.Node
 {
     [Checker(NodeConstants.NodeToolName)]
     public class NodeVersionChecker : IChecker
     {
+        private readonly ILogger<NodeVersionChecker> _logger;
+
+        public NodeVersionChecker(ILogger<NodeVersionChecker> logger)
+        {
+            _logger = logger;
+        }
+
         [NotNull]
         public IEnumerable<ICheckerMessage> CheckToolVersions(IDictionary<string, string> tools)
         {
-            if (tools.ContainsKey(NodeConstants.NodeJsName))
+            var used = tools[NodeConstants.NodeToolName];
+            var comparison = SemanticVersionResolver.CompareVersions(used, NodeScriptGeneratorOptionsSetup.NodeLtsVersion);
+            _logger.LogDebug($"SemanticVersionResolver.CompareVersions returned {comparison}");
+            if (comparison < 0)
             {
-                var used = tools[NodeConstants.NodeJsName];
-                if (SemanticVersionResolver.CompareVersions(used, NodeScriptGeneratorOptionsSetup.NodeLtsVersion) < 0)
+                return new[]
                 {
-                    return new[]
-                    {
-                        new CheckerMessage($"An outdated version of Node.js was used ({used}). Consider updating. " +
-                                           $"Versions supported in Oryx: {Constants.OryxGitHubUrl}")
-                    };
-                }
+                    new CheckerMessage($"An outdated version of Node.js was detected ({used}). Consider updating.\n" +
```

Comment:
we have a resource file on this project to put labels in. We should store those strings there so its easier to review them.